### PR TITLE
VER-1895: new repo for retry

### PR DESCRIPTION
**Pull request by YuanZHANG910**

### CODE REVIEW

File: app/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidationRetryRepository.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                                                         mongoComponent: ReactiveMongoComponent)(implicit ec: ExecutionContext)
+  extends ReactiveRepository[Retry, BSONObjectID](
+    collectionName = "personal-details-validation-retry-store",
+    mongo = mongoComponent.mongoConnector.db,
+    domainFormat = Retry.format,
+    idFormat = ReactiveMongoFormats.objectIdFormats) with RetryMongoIndexes[Retry, BSONObjectID] {
+
+  override val ttl: Long = config.collectionTtl.getSeconds
+
+  def recordAttempt(maybeCredId: String): Future[Done] = {
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidationRetryRepository.scala:68]: Why is this called "maybeCredId"?  It's not an option...

File: app/uk/gov/hmrc/personaldetailsvalidation/RetryMongoIndexes.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  val ttl: Long
+
+  override def ensureIndexes(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
+
+    import reactivemongo.bson.DefaultBSONHandlers._
+
+    val indexes: Future[List[Index]] = collection.indexesManager.list()
+
+    def ensureTtlIndex(implicit ec: ExecutionContext): Future[Seq[Boolean]] = {
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/personaldetailsvalidation/RetryMongoIndexes.scala:43]: I am a little confused - why do both collections have the same indexes?  The new collection doesn't have "createdAt" field, right?

Comment (stevenmallen) [inline app/uk/gov/hmrc/personaldetailsvalidation/RetryMongoIndexes.scala]: But it *should* have a (unique) index on credentialId I would think?

**Review by cjrudd — APPROVED** [review]
lgtm

**Pull request merged**